packages/react-form-renderer/src/files/ui-state-reducer.js

```diff
@@ -0,0 +1,92 @@
+const validAttributes = [
+  'visible',
+  'disabled',
+  'hidden',
+  'enabled',
+  'icon',
+  'help',
+  'colorBar',
+  'required',
+  'max',
+  'min',
+];
+
+//enabled and hidden are not used internally, but accepted as input from the conditions object
+const inversedTypes = {
+  enabled: 'disabled',
+  hidden: 'visible',
+};
+
+const uiStateReducer = (state, action) => {
+  switch (action.type) {
+    case 'addUIState': {
+      const {source, uiState} = action;
+
+      Object.entries(uiState).forEach(([key, item]) => {
+        //Create the item object for this item if it doesn't exist
+        if (!state.fields[key]) state.fields[key] = {};
```

Comment:
you should probably create a new copy state object before mutating the variable. We might encounter issues with async rendering and invalid state.

Reply:
Fixed in 12ce210. I'm not used to handle immutability manually in plain JS. Please check the code so I got it right.